lib/connection_pool.rb

```diff
@@ -92,7 +105,14 @@ def checkin
     nil
   end
 
+  # Removes a connection from the pool and makes the space available again
+  # connection may not currently be checked out of the queue.
+  def remove_connection(conn)
```

Comment:
If this is called during a `with` block, or before `checkin` is called the connection will end up back in the queue.

This will not close the connection